pkg/api/metrics.go

```diff
@@ -6,6 +6,7 @@ import (
 )
 
 type Metrics struct {
+	LastRequestUnixNano int64
```

Comment:
we should either add a comment explaining that this needs to be the first field due to atomic-access requirements, or we should allocate it in it's own heap allocation